src/nvim/msgpack_rpc/channel.c

```diff
@@ -521,8 +521,19 @@ void rpc_close(Channel *channel)
   }
 }
 
+static void exit_delay_cb(uv_timer_t *handle)
+{
+  uv_timer_stop(&main_loop.exit_delay_timer);
+  multiqueue_put(main_loop.fast_events, exit_event, 0);
+}
+
 static void exit_event(void **argv)
 {
+  if (exit_need_delay) {
+    uv_timer_start(&main_loop.exit_delay_timer, exit_delay_cb, 0, 0);
```

Comment:
is the timer needed to wake the event loop?

Reply:
`multiqueue_push()`, `multiqueue_process_events()` and `multiqueue_remove()` are implemented as follows:
https://github.com/neovim/neovim/blob/ed429c00d76414d07d7c7129f65ddf9be68e24f4/src/nvim/event/multiqueue.c#L235
https://github.com/neovim/neovim/blob/ed429c00d76414d07d7c7129f65ddf9be68e24f4/src/nvim/event/multiqueue.c#L150-L151
https://github.com/neovim/neovim/blob/ed429c00d76414d07d7c7129f65ddf9be68e24f4/src/nvim/event/multiqueue.c#L219
Therefore, the easiest way to queue a delayed event without causing an infinite loop in the queue currently being processed is to use a timer. Of course, you could safely do that by using `QUEUE_INSERT_HEAD` and `QUEUE_FOREACH`, but since that process is currently only needed for this PR, I don't think the cost of implementing it is worth it.